src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.IRawElementProviderFragmentVtbl.cs

```diff
@@ -0,0 +1,176 @@
+                    *pRetVal = result is null ? IntPtr.Zero : Marshal.GetIUnknownForObject(result);
+                    return HRESULT.S_OK;
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex);
+                    return (HRESULT)ex.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static HRESULT get_BoundingRectangle(IntPtr thisPtr, UiaCore.UiaRect* pRetVal)
+            {
+                if (pRetVal == null)
+                {
+                    return HRESULT.E_INVALIDARG;
+                }
+
+                var instance = ComInterfaceDispatch.GetInstance<UiaCore.IRawElementProviderFragment>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    *pRetVal = instance.BoundingRectangle;
+                    return HRESULT.S_OK;
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex);
+                    return (HRESULT)ex.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static HRESULT GetEmbeddedFragmentRoots(IntPtr thisPtr, IntPtr* pRetVal)
+            {
+                if (pRetVal == null)
+                {
+                    return HRESULT.E_INVALIDARG;
+                }
+
+                var instance = ComInterfaceDispatch.GetInstance<UiaCore.IRawElementProviderFragment>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    var objects = instance.GetEmbeddedFragmentRoots();
+                    if (objects == null)
+                    {
+                        *pRetVal = IntPtr.Zero;
+                        return HRESULT.S_OK;
+                    }
+
+                    return HRESULT.E_NOTIMPL;
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex);
+                    return (HRESULT)ex.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static unsafe HRESULT GetRuntimeId(IntPtr thisPtr, IntPtr* pRetVal)
+            {
+                if (pRetVal == null)
+                {
+                    return HRESULT.E_INVALIDARG;
+                }
+
+                var instance = ComInterfaceDispatch.GetInstance<UiaCore.IRawElementProviderFragment>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    var result = instance.GetRuntimeId();
+                    if (result == null)
+                    {
+                        *pRetVal = IntPtr.Zero;
+                        return HRESULT.S_OK;
+                    }
+
+                    var array = Oleaut32.SafeArrayCreateVector(Ole32.VARENUM.I4, 0, (uint)result.Length);
+                    fixed (int* pResult = result)
+                    {
+                        for (var i = 0; i < result.Length; i++)
```

Comment:
For the integral types we always use the type
```suggestion
                        for (int i = 0; i < result.Length; i++)
```